CRCTray.cs

```diff
@@ -237,28 +237,37 @@ async private void StartMenu_Click(object sender, EventArgs e)
             TaskHelpers.TryTask(Tasks.SendTelemetry, Actions.ClickStart).NoAwait();
 
             // Check using get-config if pullSecret is configured
-            var pullsecret = await TaskHelpers.TryTask(Tasks.GetPullSecret);
-            if (!pullsecret)
+            try
             {
-                var pullSecretForm = new PullSecretForm();
-                pullSecretForm.ShowDialog();
-                var pullSecretContent = pullSecretForm.PullSecret;
-                if (pullSecretContent == String.Empty)
-                {
-                    TrayIcon.NotifyWarn(@"No Pull Secret was provided. Cannot start cluster without pull secret.");
-                    return;
-                }
-
-                TaskHelpers.TryTask(Tasks.SendTelemetry, Actions.EnterPullSecret).NoAwait();
 
-                // Store pullsecret; returns false is not able to
-                var stored = await TaskHelpers.TryTask(Tasks.SetPullSecret, pullSecretContent);
-                if(!stored)
+                var pullsecret = await TaskHelpers.TryTaskWithRetry(Tasks.GetPullSecret);
+                if (!pullsecret)
                 {
-                    TrayIcon.NotifyError(@"Pull-secret not stored. Please check if valid format is used.");
-                    return;
-                }
+                    var pullSecretForm = new PullSecretForm();
+                    pullSecretForm.ShowDialog();
+                    var pullSecretContent = pullSecretForm.PullSecret;
+                    if (pullSecretContent == String.Empty)
+                    {
+                        TrayIcon.NotifyWarn(@"No Pull Secret was provided. Cannot start cluster without pull secret.");
+                        return;
+                    }
+
+                    TaskHelpers.TryTask(Tasks.SendTelemetry, Actions.EnterPullSecret).NoAwait();
+
+                    // Store pullsecret; returns false is not able to
+                    var stored = await TaskHelpers.TryTask(Tasks.SetPullSecret, pullSecretContent);
+                    if (!stored)
+                    {
+                        TrayIcon.NotifyError(@"Pull-secret not stored. Please check if valid format is used.");
+                        return;
+                    }
 
+                }
+            }
+            catch (RetryableTaskFailedException ex)

```

Comment:
This try-catch-block wraps too much. Only the `TryTaskWithRetry` needs this. The `return` will break the function anyway, so it is not like you prevent code execution.

Reply:
Its because of the scope of the tray block, we need to use the result of the task, so wanted to avoid the declaration of `pullsecret` outside the tray-catch.

Reply:
perhaps better to declare as:

```cs
            bool pullsecretExists = false;
            try
            {
                pullsecretExists = await TaskHelpers.TryTaskWithRetry(Tasks.GetPullSecret);
            }
```